aws/resource_aws_lex_slot_type_test.go

```diff
@@ -0,0 +1,402 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lexmodelbuildingservice"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccLexSlotType_basic(t *testing.T) {
+	var v lexmodelbuildingservice.GetSlotTypeOutput
+	rName := "aws_lex_slot_type.test"
+	testSlotTypeID := "test_slot_type_" + acctest.RandStringFromCharSet(8, acctest.CharSetAlpha)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsLexSlotTypeNotExists(testSlotTypeID, LexSlotTypeVersionLatest),
```

Comment:
Are we sure it's possible for this state to occur? I thought if you delete the slot type all versions are deleted, so you should just be able to check for it's existence. And we don't provide a way to delete versions with this resource.

Reply:
Sorry, it's been a while since I looked at this. I believe I ran into problems if there were multiple published versions. In that case, the latest version could be deleted but older versions would not be.